contracts/test/plugin/setup/plugin-setup.ts

```diff
@@ -23,42 +32,158 @@ describe('IPluginSetup', function () {
 });
 
 describe('PluginSetup', async () => {
-  let deployer: SignerWithAddress;
-  let pluginSetup: PluginSetupMockBuild1;
+  pluginSetupBaseTests(pluginSetupFixture);
+
+  describe('prepareUpdate', async () => {
+    it('reverts when called', async () => {
+      const {pluginSetupMock} = await loadFixture(pluginSetupFixture);
+
+      const dummyDaoAddr = ADDRESS.ONE;
+      const dummyPluginAddr = ADDRESS.TWO;
+      const dummyFromBuildNumber = 123;
+      const setupPayload: IPluginSetup.SetupPayloadStruct = {
+        plugin: dummyPluginAddr,
+        currentHelpers: [],
+        data: [],
+      };
+
+      await expect(
+        pluginSetupMock.prepareUpdate(
+          dummyDaoAddr,
+          dummyFromBuildNumber,
+          setupPayload
+        )
+      )
+        .to.be.revertedWithCustomError(pluginSetupMock, 'NonUpgradeablePlugin')
+        .withArgs();
+    });
+  });
+});
+
+describe('PluginUUPSUpgradeableSetup', async () => {
+  pluginSetupBaseTests(pluginUUPSUpgradeableSetupFixture);
+
+  describe('prepareUpdate', async () => {
+    // TODO This should be the default behavior that must explicitly be overridden, which is currently not the case.
+    it('should revert when called on the initial build 1', async () => {
```

Comment:
As already discussed with @mathewmeconry, I think the default behavior should be to revert on the initial build.
This could be achieved by providing 
```solidity
/// @dev Updating to the initial build 1 is not possible because no earlier build exists.
function prepareUpdate(
    address _dao,
    uint16 _fromBuild,
    SetupPayload calldata _payload
) external pure virtual returns (bytes memory, PreparedSetupData memory) {
    (_dao, _fromBuild, _payload);
    revert InvalidUpdatePath({fromBuild: 0, thisBuild: THIS_BUILD});
}
```
as a default implementation in `PluginUpgradeableSetup`.
This makes life easier for developers creating the initial build, because they don't have to implement this function, which could be confusing for them.

A default implementation that will revert if not actively overriden is, IMO, safer and better than 
- forcing devs to implement it themselves for the first build (which could go wrong)
- providing an empty default implementation that could potentially be called by someone

Reply:
@Rekard0 FYI

Reply:
Commit 24997398eb3e2b31d9ecd0bc2815c21483a801d9 adds a reverting `prepareUpdate` default implementation.
We can revert/change it depending on the discussion outcome.

Reply:
Reverted in [8d32cb1](https://github.com/aragon/osx-commons/pull/46/commits/8d32cb164ada125357a0efec4ec7d074e11aac5d)